src/librustc_trans/debuginfo/metadata.rs

```diff
@@ -798,20 +798,54 @@ pub fn compile_unit_metadata(scc: &SharedCrateContext,
         let file_metadata = llvm::LLVMRustDIBuilderCreateFile(
             debug_context.builder, compile_unit_name, work_dir.as_ptr());
 
-        return llvm::LLVMRustDIBuilderCreateCompileUnit(
+        let unit_metadata = llvm::LLVMRustDIBuilderCreateCompileUnit(
             debug_context.builder,
             DW_LANG_RUST,
             file_metadata,
             producer.as_ptr(),
             sess.opts.optimize != config::OptLevel::No,
             flags.as_ptr() as *const _,
             0,
-            split_name.as_ptr() as *const _)
+            split_name.as_ptr() as *const _);
+
+        let cu_desc_metadata = llvm::LLVMRustMetadataAsValue(debug_context.llcontext,
+                                                             unit_metadata);
+
+        let gcov_cu_info = [
+            // Ideally we would be using the three-element form of !llvm.gcov metadata,
+            // which allows us to specify gcno/gcda files explicitly, but that's only
+            // available in LLVM 3.9+; so we rely on LLVM chopping off the extension
+            // and replacing it with gcno/gcda, instead.
+            path_to_mdstring(debug_context.llcontext,
+                             &scc.output_filenames().with_extension("gcno")),
```

Comment:
Hm perhaps this could be behind a guard for clarity? I'd be wary of an update to LLVM randomly starting to read this by accident...